spec/models/cart_spec.rb

```diff
@@ -57,5 +57,17 @@
     it '.discount_total' do
       expect(@cart.discount_total).to eq(110)
     end
+
+    it '.has_discount(item)' do
+      expect(@cart.has_discount(@giant)).to eq(true)
+    end
+
+    it '.discount' do
```

Comment:
This is where I should test logic of returning more than one discount and choosing the best one!